src/components/PostDetail/components/PostDetailInfo/components/PostDetailComment/components/PostCommentList/PostCommentList.tsx

```diff
@@ -1,7 +1,72 @@
+import { PostComment } from "@/types"
 import * as S from "./PostCommentList.Styles"
+import PostCommentItem from "./components/PostCommentItem/PostCommentItem"
+import AlertModal from "@/components/Modal/components/AlertModal/AlertModal"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+import deleteComment from "@/components/PostDetail/apis/deleteComment"
+import useModal from "@/components/Modal/hooks/useModal"
+import { useState } from "react"
+import { POST_DETAIL_MODAL_MESSAGE } from "@/constants/modalMessage"
+import { POST_DETAIL_ERROR_MESSAGE } from "@/constants/errorMessage"
 
-const PostCommentList = () => {
-  return <S.PostCommentListLayout></S.PostCommentListLayout>
+const COMMENT_DELETE_MUTATION_KEY =
+  "IT_IS_DELETE_MUTATION_KEY_672532461414612689"
+
+interface PostCommentListProps {
+  commentList: PostComment[]
+}
+
+const PostCommentList = ({ commentList }: PostCommentListProps) => {
+  const {
+    isShowModal: isShowAlert,
+    showModal: showAlert,
+    closeModal: closeAlert,
+  } = useModal()
+  const [alertMessage, setAlertMessage] = useState("")
+
+  const queryClient = useQueryClient()
+  const deleteMutation = useMutation({
+    mutationKey: [COMMENT_DELETE_MUTATION_KEY],
+    mutationFn: deleteComment,
+    onSuccess: () => {
+      queryClient.refetchQueries()
+      setAlertMessage(POST_DETAIL_MODAL_MESSAGE.COMPLETE.COMMENT_DELETE)
+      showAlert()
+      return
+    },
+    onError: () => {
+      setAlertMessage(POST_DETAIL_ERROR_MESSAGE.COMMENT.DELETE)
+      showAlert()
+    },
+  })
+
+  const handleDeleteComment = (commentId: string) => {
+    deleteMutation.mutate(commentId)
+  }
+
+  const reversedList = [...commentList].reverse()
+  return (
+    <>
+      <S.PostCommentListLayout>
+        {reversedList &&
+          reversedList.map((comment) => (
+            <PostCommentItem
+              key={comment._id}
+              comment={comment}
+              onDelete={handleDeleteComment}
+            />
+          ))}
+      </S.PostCommentListLayout>
+
+      {isShowAlert && (
```

Comment:
isShow prop을 전달해주고 있으니까 `isShowAlert` 여부는 없어도 될 것 같습니다!!
```suggestion
              <AlertModal
          alertMessage={alertMessage}
          isShow={isShowAlert}
          onClose={closeAlert}
        />
```

Reply:
해당 isShowAlert은 렌더링과정에서 false라면 모달 Component의 내부 조건 까지의 렌더링을 막기위함이었는데 의미없을려나욤?

Reply:
아하 그런 장점은 있을 것 같아요!!
근데 이렇게 하면 `isShow` prop이 아무 의미가 없어진다는 단점이 있어요.(`isShow={true} `로 주어도 아무 문제가 없더라고요!)

제 생각엔 show 여부는 `alertModal`의 `isShow`로만 판단하고 처리하는게 더 좋을 것 같은데 재영님은 어떠신가요???

추가로 아래처럼 `useModal`의 `showAlert`만 `useDeleteComment`에서 사용하고, `isShow`, `onClose`는 `AlertModal`에서 가져오면 관심사 분리가 더 잘 될 것 같아요!!

`useDeleteComment.ts`
```
const { showModal: showAlert } = useModal()
...
const DeleteCommentAlertModal = <AlertModal alertMessage={alertMessage} />

```

`AlertModal.tsx`
```
  const { isShowModal: isShowAlert, closeModal: closeAlert } = useModal()
...
  return (
    <CustomModal
      isShow={isShowAlert}
      $height={20}
      onClickAway={closeAlert}
    >
...

```

Reply:
저도 지금 테스트를 해보니 말씀주신것과 동일하게 크게 의미가 없는 방어 인 것 같네요.. 🥹 딱 Portal의 중간 조건까지만 방어해줘서... 전부 제거하겠슴돵

Reply:
전부 제거했습니다 완료